src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,21 +20,82 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/**
- * Servlet that returns some example content.
- * comments data
- */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static String TIMESTAMP = "timestamp";
+  private static String TITLE = "title";
+  private static String SENTIMENT = "sentiment";
+  private static String UPVOTES = "upvotes";
+  private static String URL = "url";
```

Comment:
nit: please make these final as well

Reply:
Typically, static final vars are first in the class, and then you can have the final / non static vars.